src/activities/mirrorgame/mirrorgame.js

```diff
@@ -0,0 +1,526 @@
+                            if(Light[k1].uniquename === "lightpath"+row+column+row1+column1 &&
+                                    Light[k1].uniquename !== "lightpath"+row+column+i+j &&
+                                    Light[k1].visible===true)
+                            {
+                                Light[k1].visible = false;
+                            }
+                        }}}
+            }
+            else if(Light[k].uniquename === "lightpath"+i+j+row+column && Light[k].visible ===true)
+            {
+                for(row1 = 0; row1 < items.rows; row1++){
+                    for(column1 = 0; column1 < items.columns; column1++){
+                        for(k1 = 0; k1 < count; k1++) {
+                            if(Light[k1].uniquename === "lightpath"+row+column+row1+column1 &&
+                                    Light[k1].uniquename !== "lightpath"+row+column+i+j &&
+                                    Light[k1].visible===true)
+                            {
+                                Light[k1].visible = false;
+                            }
+                        }}}
+            }
+        }}}
+
+  switch(mirror[i][j].pos)
+        {
+            case 1:
+            while(currLight[i][j].length!==0)
+            {
+                var x = currLight[i][j].pop()
+                x.visible = false
+            }
+            for(row = 0; row < items.rows; row++){var done = 0
+                for(column = 0; column < items.columns; column++){
+                    for(k = 0; k < count; k++) {
+
+              if(Light[k].uniquename === "lightpath"+row+column+i+j && Light[k].visible === true
+                  && row<i && column>j)
+                {
+                 for(var a =0; a < items.rows;a++) {
+                     for(var b = 0; b < items.columns; b++) {
+                        for(var c = 0; c < count ; c++) {
+                     if(Light[c].uniquename === "lightpath"+i+j+a+b && i<a && j<b)
+                        {
+                          Light[c].visible = true
+                         currLight[i][j].push(Light[c])
+                         done = 1
+                        }
+                 }}}}
+              else if(Light[k].uniquename === "lightpath"+i+j+row+column && Light[k].visible === true
+                      && row>i && column>j)
+              {
+                  for( a =0; a < items.rows;a++) {
+                      for( b = 0; b < items.columns; b++) {
+                         for( c = 0; c < count ; c++) {
+                      if(Light[c].uniquename === "lightpath"+a+b+i+j && i<a && j>b)
+                         {
+                          Light[c].visible = true;
+                         currLight[i][j].push(Light[c])
+                          done = 1
+                      }
+                  }}}
+
+
+              }}if(done)break}if(done)break}
+              break
+
+          case 2:
+              while(currLight[i][j].length!==0)
```

Comment:
what are the differences between the different cases in the switch.
Maybe some code is factorisable?